src/Element/Zap.tsx

```diff
@@ -0,0 +1,151 @@
+import "./Zap.css";
+import { useMemo } from "react";
+import { useSelector } from "react-redux";
+// @ts-expect-error
+import { decode as invoiceDecode } from "light-bolt11-decoder";
+import { bytesToHex } from "@noble/hashes/utils";
+
+//import { sha256 } from "Util";
+import { formatShort } from "Number";
+import { HexKey, TaggedRawEvent } from "Nostr";
+import Event from "Nostr/Event";
+import Text from "Element/Text";
+import ProfileImage from "Element/ProfileImage";
+import { RootState } from "State/Store";
+
+function findTag(e: TaggedRawEvent, tag: string) {
+  const maybeTag = e.tags.find((evTag) => {
+    return evTag[0] === tag
+  })
+  return maybeTag && maybeTag[1]
+}
+
+function getInvoice(zap: TaggedRawEvent) {
+  const bolt11 = findTag(zap, 'bolt11')
+  const decoded = invoiceDecode(bolt11)
+
+  const amount = decoded.sections.find((section: any) => section.name === 'amount')?.value
+  const hash = decoded.sections.find((section: any) => section.name === 'description_hash')?.value;
+
+  return { amount, hash: hash ? bytesToHex(hash) : undefined };
+}
+
+function getZapper(zap: TaggedRawEvent, dhash: string) {
+  const zapRequest = findTag(zap, 'description')
+  if (zapRequest) {
+    const rawEvent: TaggedRawEvent = JSON.parse(zapRequest);
+    if (Array.isArray(rawEvent)) {
+      // old format, ignored
+      const rawDescriptionTag = rawEvent.find(a => a[0] === 'application/nostr')
+      const rawDescription = rawDescriptionTag && rawDescriptionTag[1]
+      const request = typeof rawDescription === 'string' ? JSON.parse(rawDescription) : rawDescription
+      return request?.pubkey
+    }
+    //const metaHash = sha256(zapRequest);
```

Comment:
This is wrong, were not using this format anymore, can you revert to previous?

The `description` tag contains the zap note and the hash of this should match the description hash of the invoice